### PR TITLE
Use ttl.sh to run the operator tests

### LINKED ISSUE #11214: Re-introduce OLM tests in GitHub Actions
This is a follow-up from: #11206. See it for more details why the tests must've been removed from the CI.

### PR DESCRIPTION
**Pull request by andreaTP**
Opening in draft for visibility.

Resolves #11214

### CODE REVIEW

File: operator/scripts/prepare-olm-test.sh [hunk 1]
```diff
@@ -8,25 +8,27 @@ DOCKER_REGISTRY="$1"
 VERSION="$2"
 PREV_VERSION="$3"
 
+UUID="$4"
```
Comment (vmuzikar) [inline operator/scripts/prepare-olm-test.sh:11]: Maybe we could make the param optional for better UX when running locally?

**Review by vmuzikar — COMMENTED** [review]
@andreaTP Wow, ttl.sh might've just saved us! :) Very nice solution in comparison to local docker registry. Let's hope ttl.sh will have a long Time-to-Live and won't reach EOL anytime soon. :) But let's worry about that later.

File: .github/workflows/operator-ci.yml [hunk 2]
```diff
@@ -143,27 +147,20 @@ jobs:
...
+          eval $(minikube -p minikube docker-env)
           mvn clean package \
               -Dquarkus.container-image.build=true \
-              -Dquarkus.container-image.push=true \
-              -Dquarkus.container-image.image="${{ secrets.TEST_DOCKER_REGISTRY }}/${{ secrets.TEST_DOCKER_REPOSITORY }}/keycloak-operator:${{ steps.vars.outputs.version }}" \
-              -Dquarkus.container-image.labels."\"quay.expires-after\""="20h" \
+              -Dquarkus.container-image.image="ttl.sh/${GITHUB_SHA::6}keycloak-operator:${{ steps.vars.outputs.version }}" \
               -DskipTests
+          # TODO: try to embed the push back in mvn
+          docker push ttl.sh/${GITHUB_SHA::6}keycloak-operator:${{ steps.vars.outputs.version }}
```
[the remark this answers is not in the document]
Comment (andreaTP) [reply .github/workflows/operator-ci.yml]: it wasn't working in one of the "configuration" I was trying out, reverted now 👍

File: operator/scripts/create-olm-test-resources.sh [hunk 3]
```diff
@@ -4,6 +4,8 @@ set -euxo pipefail
 VERSION=$1
 DOCKER_REGISTRY=$2
 
+UUID=$3
```
[the remark this answers is not in the document]
Comment (andreaTP) [reply operator/scripts/create-olm-test-resources.sh]: done

File: operator/scripts/prepare-olm-test.sh (hunk 1 above)
Comment (andreaTP) [reply operator/scripts/prepare-olm-test.sh]: done

**Review by vmuzikar — APPROVED** [review]
LGTM, thanks!

**Review by abstractj — APPROVED** [review]

**Pull request merged**